src/cli/src/lib.rs

```diff
@@ -32,21 +32,23 @@ impl CLI {
                     .long("memory")
                     .required(true)
                     .takes_value(true)
+                    .default_value("size_mib=256")
```

Comment:
The docs will need to be updated with some info about the default values and, if we change the separators (or other things), those will have to be updated there too.

Reply:
The default value is already printed by the default help.

This is how the help looks like with these changes:
```bash

USAGE:
    target/debug/vmm-reference --kernel <kernel> --memory <memory> --vcpu <vcpu>

FLAGS:
    -h, --help       
            Prints help information

    -V, --version    
            Prints version information


OPTIONS:
        --kernel <kernel>    
            Kernel configuration.
            	Format: "path=<string> [cmdline=<string> himem_start=<u64>]"
        --memory <memory>    
            Guest memory configuration.
            	Format: "size_mib=<u32>" [default: size_mib=256]
        --vcpu <vcpu>        
            vCPU configuration.
            	Format: "num=<u8>" [default: num=1]

```

For the kernel configuration I didn't find a good way of printing the help and also denote that some of the config options are optional.

Reply:
Do you mean other documents rather than the help?

Reply:
Yep, sorry, I was referring to DESIGN.md and getting-started, the command examples could be updated, for example the one from [here](https://github.com/rust-vmm/vmm-reference/blob/master/docs/DESIGN.md#user-interface), maybe the `Inputs` from [here](https://github.com/rust-vmm/vmm-reference/blob/master/docs/DESIGN.md#functionalities) too (with mandatory/not mandatory or their defaults for example).

Reply:
Okappa, will update.

Reply:
I've updated the readme and added 2 more examples. For the usage example i chose the most basic example.

In the design document, I chose to remove the "user defined" mention because otherwise we would end up with duplicated information in two places which I believe makes things a bit harder to maintain.

Reply:
:+1: These [lines](https://github.com/rust-vmm/vmm-reference/blame/master/docs/DESIGN.md#L158-L161) and the examples from [here](https://github.com/rust-vmm/vmm-reference/blob/master/docs/getting-started.md#putting-it-all-together) will also need an update.

Also, totally unrelated to this PR, just noticed that we didn't update the documentation related to all the changes introduced in the bzimage PR :(.

Reply:
I've updates those as well, but realized that the documentation is behind by a large margin. We need to go back and update it. For the programmatic interface, I would wait a bit more, and I am also considering not mentioning function names & such because those are always in flux. For code level documentation, people can do a cargo doc --open on the vmm crate, so we could encourage them to do that instead (or even host something like that). Then we can move the documentation in the crate itself so that the examples are also built which increases the changes of the documentation to stay up to date. @aghecenco what do you think?

In the meantime, I've open this issue: #39